desktop/src/main/java/bisq/desktop/components/paymentmethods/CapitualForm.java

```diff
@@ -0,0 +1,126 @@
+import bisq.core.payment.PaymentAccount;
+import bisq.core.payment.payload.CapitualAccountPayload;
+import bisq.core.payment.payload.PaymentAccountPayload;
+import bisq.core.util.coin.CoinFormatter;
+import bisq.core.util.validation.InputValidator;
+
+import bisq.common.util.Tuple2;
+
+import org.apache.commons.lang3.StringUtils;
+
+import javafx.scene.control.Label;
+import javafx.scene.layout.FlowPane;
+import javafx.scene.layout.GridPane;
+
+import static bisq.desktop.util.FormBuilder.addCompactTopLabelTextField;
+import static bisq.desktop.util.FormBuilder.addCompactTopLabelTextFieldWithCopyIcon;
+import static bisq.desktop.util.FormBuilder.addTopLabelFlowPane;
+
+public class CapitualForm extends PaymentMethodForm {
+    private final CapitualAccount capitualAccount;
+    private final CapitualValidator capitualValidator;
+    private InputTextField accountNrInputTextField;
+
+    public static int addFormForBuyer(GridPane gridPane, int gridRow,
+                                      PaymentAccountPayload paymentAccountPayload) {
+        addCompactTopLabelTextFieldWithCopyIcon(gridPane, ++gridRow, Res.get("payment.capitual.cap"),
+                ((CapitualAccountPayload) paymentAccountPayload).getAccountNr());
+        return gridRow;
+    }
+
+    public CapitualForm(PaymentAccount paymentAccount, AccountAgeWitnessService accountAgeWitnessService, CapitualValidator capitualValidator,
+                        InputValidator inputValidator, GridPane gridPane, int gridRow, CoinFormatter formatter) {
+        super(paymentAccount, accountAgeWitnessService, inputValidator, gridPane, gridRow, formatter);
+        this.capitualAccount = (CapitualAccount) paymentAccount;
+        this.capitualValidator = capitualValidator;
+    }
+
+    @Override
+    public void addFormForAddAccount() {
+        gridRowFrom = gridRow + 1;
+
+        accountNrInputTextField = FormBuilder.addInputTextField(gridPane, ++gridRow, Res.get("payment.capitual.cap"));
+        accountNrInputTextField.setValidator(capitualValidator);
+        accountNrInputTextField.textProperty().addListener((ov, oldValue, newValue) -> {
+            capitualAccount.setAccountNr(newValue);
+            updateFromInputs();
+        });
+
+        addCurrenciesGrid(true);
+        addLimitations(false);
+        addAccountNameTextFieldWithAutoFillToggleButton();
+    }
+
+    private void addCurrenciesGrid(boolean isEditable) {
+        final Tuple2<Label, FlowPane> labelFlowPaneTuple2 = addTopLabelFlowPane(gridPane, ++gridRow, Res.get("payment.supportedCurrencies"), 0);
+
+        FlowPane flowPane = labelFlowPaneTuple2.second;
+
+        if (isEditable)
+            flowPane.setId("flow-pane-checkboxes-bg");
+        else
+            flowPane.setId("flow-pane-checkboxes-non-editable-bg");
+
+        CurrencyUtil.getAllCapitualCurrencies().stream().forEach(e ->
+                fillUpFlowPaneWithCurrencies(isEditable, flowPane, e, capitualAccount));
```

Comment:
You can use `forEach` on the list itself
```suggestion
        CurrencyUtil.getAllCapitualCurrencies().forEach(e ->
                fillUpFlowPaneWithCurrencies(isEditable, flowPane, e, capitualAccount));
```